program/rust/src/rust_oracle.rs

```diff
@@ -22,6 +96,42 @@ pub fn update_version(
     accounts: &Vec<AccountInfo>,
     instruction_data: &[u8],
 ) -> OracleResult {
-    panic!("Need to merge fix to pythd in order to implement this");
+    //read account info
+    let pyth_acc_info_size = size_of::<pc_acc_t>();
+    let mut pyth_acc_info =
+        pc_acc_t::try_from_slice(&accounts[1].try_borrow_data()?[..pyth_acc_info_size])?;
+
+    //check magic number
+    if pyth_acc_info.magic_ != PC_MAGIC {
+        //TODO: once we settle on error stuff fix this
+        return Err(OracleError::IncorrectMagicNumber.into());
+    }
+
+    //update the accounts depending on type
+    match pyth_acc_info.type_ {
+        PC_ACCTYPE_PRICE => upgrade_price_account(program_id, &accounts, instruction_data)?,
+        _ => (),
+    }
+
+    //I do not think that we need to have PC_VERSION dependent logic now, but here seems like a
+    // good place to do such a thing
+
+    //update the account info
+    pyth_acc_info.ver_ = PC_VERSION;
```

Comment:
I think it's fine leaving everything in version 2. After all, layouts are not changing. Do you have another opinion.